torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -111,29 +127,83 @@ def create_subclass_meta(
 # a list of tensors that we would then need to concat together.
 # Instead, we specialize the logic for the inference vs. joint graph case.
 # NOTE: this function is hot, since we unwrap tensor subclass inputs at runtime
-def unwrap_tensor_subclasses(wrapped_args, *, is_joint_structure: bool):
-    def concat_inner_tensors_from_subclasses(xs):
-        xs_inner = []
-        for x in xs:
+
+
+def filter_symints(size: Union[Size, List[Union[int, SymInt]]]) -> List[SymInt]:
+    return [s for s in size if isinstance(s, SymInt)]
+
+
+def unwrap_tensor_subclasses(
+    wrapped_args: List[Union[Tensor, int]],
+    *,
+    append_symints: bool,
+    subclass_metas: Optional[List[Union[int, SubclassCreationMeta]]] = None,
+):
+    xs_inner: List[Union[int, Tensor, SymInt]] = []
+    sizes: List[SymInt] = []
+    for x in wrapped_args:
+        if is_traceable_wrapper_subclass(x):
+            xs_inner.extend(get_plain_tensors(typing.cast(Tensor, x)))
+            sizes.extend(filter_symints(x.size()))
+        else:
+            xs_inner.append(x)
+
+    if append_symints:
+        xs_inner.extend(sizes)
+    return xs_inner
+
+
+def runtime_unwrap_tensor_subclasses(
+    wrapped_args: List[Union[Tensor, int]],
+    *,
+    append_symints: bool,
+    subclass_metas: List[Union[int, SubclassCreationMeta]],
+):
+    # re-use the non-runtime version
+    xs_inner = unwrap_tensor_subclasses(
+        wrapped_args, append_symints=False, subclass_metas=None
```

Comment:
it's a bit confusing to me that we have a separate, dedicated util to unwrap subclasses for the runtime path, but it actual just calls `unwrap_tensor_subclases` directly. It also passes in `append_symints=False,` but then has its own `append_symints` flag.

Any reason we can't just call `unwrap_tensor_subclasses()` directly, with the right set of kwargs?

Reply:
As per your suggestion in a previous comment, I've unified both functions into `unwrap_tensor_subclasses()`.